src/System.Text.Json/tests/Utf8JsonReaderTests.MultiSegment.cs

```diff
@@ -142,6 +142,259 @@ private static void ReadDoubleHelper(ref Utf8JsonReader jsonReader, int expected
+            }
+
+            for (int firstSplit = 0; firstSplit < utf8.Length; firstSplit++)
+            {
+                for (int secondSplit = firstSplit; secondSplit < utf8.Length; secondSplit++)
+                {
+                    sequence = JsonTestHelper.CreateSegments(utf8, firstSplit, secondSplit);
+                    jsonReader = new Utf8JsonReader(sequence);
+                    InvalidReadNumberHelper(ref jsonReader, expectedBytePositionInLine, numberOfSuccessfulReads, expectedConsumed, invalidChar);
+                }
+            }
+        }
+
+        private static void InvalidReadNumberHelper(ref Utf8JsonReader jsonReader, int expectedBytePositionInLine, int numberOfSuccessfulReads, int expectedConsumed, char invalidChar)
+        {
+            try
+            {
+                for (int i = 0; i < numberOfSuccessfulReads; i++)
+                {
+                    Assert.True(jsonReader.Read());
+
+                    long tokenLength = jsonReader.HasValueSequence ? jsonReader.ValueSequence.Length : jsonReader.ValueSpan.Length;
+                    Assert.Equal(expectedConsumed, tokenLength);
+                    Assert.Equal(0, jsonReader.TokenStartIndex);
+                    Assert.Equal(expectedConsumed, jsonReader.BytesConsumed - jsonReader.TokenStartIndex);
+                }
+                Assert.False(jsonReader.Read());
+                Assert.True(false, "Expected JsonException was not thrown for incomplete/invalid JSON payload reading.");
+            }
+            catch (JsonException ex)
+            {
+                Assert.Equal(0, ex.LineNumber);
+                Assert.Equal(expectedBytePositionInLine, ex.BytePositionInLine);
+                Assert.Equal(expectedConsumed, jsonReader.BytesConsumed);
+                Assert.Contains($"'{invalidChar}'", ex.Message);
+            }
+        }
+
+        [Theory]
+        [InlineData("{\"\\\"\"", 5, 2, 5, "\"", 2)]
+        [InlineData("\"\\\"\"}", 4, 1, 4, "\"", 2)]
+        [InlineData("\"\\\"\"5", 4, 1, 4, "\"", 2)]
+        [InlineData("\"\\\"\":", 4, 1, 4, "\"", 2)]
+        [InlineData("\"\\\"\",", 4, 1, 4, "\"", 2)]
+        [InlineData("\"abc\\t\"}", 7, 1, 7, "abc\t", 5)]
+        [InlineData("\"abc\\u0061\"}", 11, 1, 11, "abca", 9)]
+        [InlineData("\"abc\\u0061abc\"}", 14, 1, 14, "abcaabc", 12)]
+        [InlineData("\"abc\\t\\\"\"}", 9, 1, 9, "abc\t\"", 7)]
+        [InlineData("\"abc\\n\"}", 1, 1, 7, "abc\n", 5, 1)]
+        [InlineData("\"abc\\na\"}", 2, 1, 8, "abc\na", 6, 1)]
+        [InlineData("\"abc\\u0061X\"}", 12, 1, 12, "abcaX", 10)]
+        [InlineData("\"\\u0061\"}", 8, 1, 8, "a", 6)]
+        public static void ValidStringWithinInvalidJsonVariousSegmentSizes(string input, int expectedBytePositionInLine, int numberOfSuccessfulReads, int expectedConsumed, string expectedStr, int expectedTokenLength, int expectedLineNumber = 0)
```

Comment:
I wonder if this theory is complex enough to warrant a dedicated input type and a MemberData generator.  I certainly have trouble remembering what 5,2,5 vs 4,1,4 means.

Reply:
(This really applies to all the new tests)

Reply:
In the interest of time, I am going to leave things as is. We have existing `MemberData`/`InlineData` that follows the current structure.